sdk/src/proto/network.rs

```diff
@@ -1,125 +1,234 @@
 #[derive(Clone, PartialEq, ::prost::Message)]
-pub struct GetProofStatusRequest {
-    #[prost(string, tag = "1")]
+pub struct ClaimProofRequest {
+    /// The nonce for the account.
+    #[prost(uint64, tag = "1")]
+    pub nonce: u64,
+    /// The proof identifier.
+    #[prost(string, tag = "2")]
     pub proof_id: ::prost::alloc::string::String,
+    /// The signature of the message.
+    #[prost(bytes = "vec", tag = "3")]
+    pub signature: ::prost::alloc::vec::Vec<u8>,
 }
+/// The response for claiming a proof, giving identifiers for the location to retrieve the program
+/// and stdin, as well as the the location to upload the proof.
 #[derive(serde::Serialize, serde::Deserialize)]
 #[allow(clippy::derive_partial_eq_without_eq)]
 #[derive(Clone, PartialEq, ::prost::Message)]
-pub struct GetProofStatusResponse {
-    #[prost(enumeration = "ProofStatus", tag = "1")]
-    pub status: i32,
+pub struct ClaimProofResponse {
+    /// The artifact identifier for the program location.
+    #[prost(string, tag = "1")]
+    pub program_artifact_id: ::prost::alloc::string::String,
+    /// The artifact identifier for the stdin location.
     #[prost(string, tag = "2")]
-    pub result_get_url: ::prost::alloc::string::String,
-    #[prost(uint32, tag = "3")]
-    pub stage: u32,
-    #[prost(uint32, tag = "4")]
-    pub total_stages: u32,
-    #[prost(string, tag = "5")]
-    pub stage_name: ::prost::alloc::string::String,
-    #[prost(uint32, optional, tag = "6")]
-    pub stage_progress: ::core::option::Option<u32>,
-    #[prost(uint32, optional, tag = "7")]
-    pub stage_total: ::core::option::Option<u32>,
+    pub stdin_artifact_id: ::prost::alloc::string::String,
+    /// The artifact identifier for the proof location.
+    #[prost(string, tag = "3")]
+    pub proof_artifact_id: ::prost::alloc::string::String,
+}
+/// The request to fulfill a proof. Should only be called after the proof has been uploaded. Only
+/// callable by the prover that claimed the proof previously.
+#[derive(serde::Serialize, serde::Deserialize)]
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct FulfillProofRequest {
+    /// The nonce for the account.
+    #[prost(uint64, tag = "1")]
+    pub nonce: u64,
+    /// The proof identifier.
+    #[prost(string, tag = "2")]
+    pub proof_id: ::prost::alloc::string::String,
+    /// The signature of the message.
+    #[prost(bytes = "vec", tag = "3")]
+    pub signature: ::prost::alloc::vec::Vec<u8>,
+}
+/// The response for fulfilling a proof, empty on success.
+#[derive(serde::Serialize, serde::Deserialize)]
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct FulfillProofResponse {
+    /// The amount of seconds between proof claim and fulfillment.
+    #[prost(uint64, tag = "1")]
+    pub proving_seconds: u64,
 }
+/// The request to relay a proof through the NetworkGateway on a given chain. Should only be called
```

Comment:
nit: Should only -> Returns an error if the proof has not been fulfilled yet

Reply:
added this to client.rs